x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityCcrIT.java

```diff
@@ -0,0 +1,364 @@
+            responseMap.forEach((k, v) -> assertThat(k, v, is(true)));
+
+            // Ensure data is replicated
+            verifyReplicatedDocuments(4L, followIndexName);
+
+            assertFollowerInfo(followIndexName, "leader-index", "active");
+            assertFollowerStats(followIndexName);
+
+            // unfollow and then follow and then index a few docs in leader index:
+            pauseFollow(followIndexName);
+            assertFollowerInfo(followIndexName, "leader-index", "paused");
+            resumeFollow(followIndexName);
+            final Request bulkRequest = new Request("POST", "/_bulk?refresh=true");
+            bulkRequest.setJsonEntity(Strings.format("""
+                { "index": { "_index": "leader-index" } }
+                { "name": "doc-5" }
+                { "index": { "_index": "leader-index" } }
+                { "name": "doc-6" }
+                """));
+            assertOK(performRequestAgainstFulfillingCluster(bulkRequest));
+            verifyReplicatedDocuments(6L, followIndexName);
+
+            pauseFollow(followIndexName);
+            closeIndex(followIndexName);
+            unfollow(followIndexName);
+            assertNoFollowerInfo(followIndexName);
+            final var e = expectThrows(ResponseException.class, () -> resumeFollow(followIndexName));
+            assertThat(e.getMessage(), containsString("follow index [" + followIndexName + "] does not have ccr metadata"));
+        }
+
+        // query cluster error cases - no privileges
+        {
+
+            final Request putCcrRequest = new Request("PUT", "/follower-index-2/_ccr/follow?wait_for_active_shards=1");
+            putCcrRequest.setJsonEntity("""
+                {
+                  "remote_cluster": "my_remote_cluster",
+                  "leader_index": "private-index"
+                }""");
+            final ResponseException e = expectThrows(ResponseException.class, () -> performRequestWithCcrUser(putCcrRequest));
+            assertThat(e.getResponse().getStatusLine().getStatusCode(), equalTo(403));
+            assertThat(e.getMessage(), containsString("insufficient privileges to follow index [private-index]"));
+        }
+
+        // query cluster error cases - aliases not supported
+        {
+            final Request putCcrRequest = new Request("PUT", "/follower-index-3/_ccr/follow?wait_for_active_shards=1");
+            putCcrRequest.setJsonEntity("""
+                {
+                  "remote_cluster": "my_remote_cluster",
+                  "leader_index": "shared-alias"
+                }""");
+            final ResponseException e = expectThrows(ResponseException.class, () -> performRequestWithCcrUser(putCcrRequest));
+            assertThat(e.getResponse().getStatusLine().getStatusCode(), equalTo(400));
+            assertThat(e.getMessage(), containsString("cannot follow [shared-alias], because it is a ALIAS"));
```

Comment:
Add a negative test case specifically for aliases since the indices action has an implicit assumption that the permission must be granted on the concrete indices.